pkg/upgrade/control_plane.go

```diff
@@ -1042,6 +1059,14 @@ func (u *ControlPlaneUpgrader) resourceExists(ref v1.ObjectReference) (*unstruct
 	return obj, nil
 }
 
+func patchMachine(machine *clusterv1.Machine, patch jsonpatch.Patch) (*clusterv1.Machine, error) {
```

Comment:
I would remove this function and place the type assertion (cast) after you apply the patch above